src/test/java/no/unit/nva/utils/MockHttpClient.java

```diff
@@ -0,0 +1,93 @@
+package no.unit.nva.utils;
+
+import java.net.Authenticator;
+import java.net.CookieHandler;
+import java.net.ProxySelector;
+import java.net.http.HttpClient;
+import java.net.http.HttpRequest;
+import java.net.http.HttpResponse;
+import java.net.http.HttpResponse.BodyHandler;
+import java.net.http.HttpResponse.PushPromiseHandler;
+import java.time.Duration;
+import java.util.Optional;
+import java.util.concurrent.CompletableFuture;
+import java.util.concurrent.Executor;
+import javax.net.ssl.SSLContext;
+import javax.net.ssl.SSLParameters;
+
+public class MockHttpClient<R> extends HttpClient {
```

Comment:
This is a lot of code. Normal mocking wasn't sufficient?

Reply:
I am not sure how to create a generic Mock.